docs/doxygen/pages.dox

```diff
@@ -19,34 +19,165 @@ Feature of HTTP/1.1 not supported in this library:
 
 <table>
     <tr>
-        <td colspan="4"><center><b>Code size of HTTP Client library files (sizes generated with [GCC for ARM Cortex-M toolchain](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads/9-2020-q2-update))</b></center></td>
+        <td colspan="7"><center><b>Code size of HTTP Client library files (sizes generated with [GCC for ARM Cortex-M toolchain](https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-rm/downloads/9-2020-q2-update))</b></center></td>
     </tr>
     <tr>
-        <td><b>File</b></td>
-        <td><b>No Optimization (asserts enabled)</b></td>
-        <td><b>With -O1 Optimization</b></td>
-        <td><b>With -Os Optimization</b></td>
+        <td><b>File ↓</b></td>
+        <td colspan="2"><b>No Optimization (asserts enabled)</b></td>
+        <td colspan="2"><b>With -O1 Optimization</b></td>
+        <td colspan="2"><b>With -Os Optimization</b></td>
     </tr>
     <tr>
-        <td>[<FIXME: library source, add more rows for more source>]</td>
-        <td>[<FIXME: Code size>]</td>
-        <td>[<FIXME: Code size>]</td>
-        <td>[<FIXME: Code size>]</td>
+        <td><b>Segment →</b></td>
+        <td><b>text</b></td>
+        <td><b>data</b></td>
+        <td><b>text</b></td>
+        <td><b>data</b></td>
+        <td><b>text</b></td>
+        <td><b>data</b></td>
     </tr>
     <tr>
-        <td><b>Total estimates</b></td>
-        <td>[<FIXME: Total code size>]</td>
-        <td>[<FIXME: Total code size>]</td>
-        <td>[<FIXME: Total code size>]</td>
+        <td>core_http_client.c</td>
+        <td>15.3K</td>
+        <td>0</td>
+        <td>9.6K</td>
+        <td>0</td>
+        <td>8.2K</td>
+        <td>0</td>
+    </tr>
+    <tr>
+        <td>http_parser.c</td>
+        <td>38.5K</td>
+        <td>412</td>
+        <td>24.9K</td>
+        <td>4</td>
+        <td>20.8K</td>
+        <td>4</td>
+    </tr>
+    <tr>
+        <td><b>Total estimates →</b></td>
+        <td>53.8K</td>
+        <td>412</td>
+        <td>34.4K</td>
+        <td>4</td>
+        <td>29.1K</td>
+        <td>4</td>
     </tr>
 </table>
  */
 
 /**
 @page http_design Design
-HTTP Client Library Design
+HTTP Client Library Architecture and Design
+
+This HTTP client library implements a subset of the HTTP/1.1 protocol. It is
+optimized for resource constrained devices and does not allocate any memory.
```

Comment:
```suggestion
optimized for resource constrained devices and does not allocate any memory on the heap.
```

Reply:
I am going to put "dynamically allocate any memory" instead. THanks